src/gsSolver/gsPatchPreconditionersCreator.hpp

```diff
@@ -324,10 +324,10 @@ typename gsPatchPreconditionersCreator<T>::OpUPtr gsPatchPreconditionersCreator<
     gsMatrix<T> ev;
 
     // Now, setup the Q's and update the D's
-    for ( index_t i=0; i<d; ++i )
+    for ( index_t i=d-1; i>=0; --i )
     {
         // Solve generalized eigenvalue problem
-        ges.compute(local_stiff[i], local_mass [i], Eigen::ComputeEigenvectors);
+        ges.compute(local_stiff[i], local_mass[i], Eigen::ComputeEigenvectors);
```

Comment:
`d-1-i` or `i`?

Reply:
yes, indeed. i fix this a bit differently.